### PR TITLE
docs: Document new Operate API endpoints, fields and limitations (including migration)

### PR DESCRIPTION
**Pull request by johanwelgemoed**
## What is the purpose of the change

Closes https://github.com/camunda/camunda-platform-docs/issues/1822

Changes to API documentation:
- Added additional column (Notes) to endpoint table
- Added new endpoints with note indicating it is new
- Added additional fields to existing endpoints with notes and limitations

Added a notes section to update guide page, indicating preferred upgrade path and current migration limitation.

## When should this change go live?

To be included in documentation as part of 8.2 release.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by akeller — COMMENTED** [review]
Quick note here regarding update vs. upgrade - https://camunda.slack.com/archives/C026U8GBNSW/p1678732380855269

**Review by christinaausley — APPROVED** [review]
Technical review in latest commit 👍

**Comment by johanwelgemoed** [comment]
@christinaausley I corrected the wording from upgrade to update as per the comment from Amara. Would you prefer a seperate PR for that or should I just amend this one?

File: docs/apis-clients/operate-api/index.md [hunk 1]
```diff
@@ -89,25 +89,27 @@ curl -b cookie.txt -X POST 'http://localhost:8080/v1/process-definitions/search'
...
+| `GET /v1/process-definitions/{key}/xml`          |                                Get process definition by key as XML |                                                                                                                                                                                                                                                        |
+| **Process instances**                            |                                                                     |                                                                                                                                                                                                                                                        |
+| `POST /v1/process-instances/search`              |                                        Search for process instances | New fields added: <br/>`flowNodeId`<br/>`flowNodeName`<br/>`processDefinitionKey` <br/><br/>**Warning**<br/>1. New fields could break deserialization, so ignore fields not used.<br/>2. processDefinitionKey only available from version 8.1.8 onward |
```
Comment (sdorokhova) [inline docs/apis-clients/operate-api/index.md:99]: @johanwelgemoed do you think we could explicitly mention that `processDefinitionKey` will be empty for data existed before upgrade to 8.1.8?

**Review by sdorokhova — DISMISSED** [review]

**Comment by akeller** [comment]
> @christinaausley I corrected the wording from upgrade to update as per the comment from Amara. Would you prefer a seperate PR for that or should I just amend this one?

You can amend this one 🙂 thank you @johanwelgemoed!

**Review by christinaausley — APPROVED** [review]

**Pull request merged**